anycloud/cli/src/deploy/mod.rs

```diff
@@ -1318,8 +1318,10 @@ where
       .into_iter()
       .filter(|line| !line.is_empty())
       .for_each(|line| {
-        if let Some(last_line) = lines.get(lines.len() - 1) {
-          sp.println(last_line);
+        if lines.len() > 0 {
+          if let Some(last_line) = lines.get(lines.len() - 1) {
+            sp.println(last_line);
+          }
         }
         sp.set_message(line);
```

Comment:
If there are multiple new entries in `new_lines` this is gonna print `last_line` multiple times. I wouldn't mind if this fix as currently written was before the functional stuff instead of inside of it?

Reply:
> > > This fixes the crash, but doesn't look like it behaves correctly?
> > > If the old `lines.len()` is `0`, it should be printing _all_ of the new lines?
> > 
> > 
> > maybe @cdmistman could confirm, but what I understand is that if `lines` has something in it it should print the last one, if it's empty no need to do anything. The new lines are still being printed.
> 
> So `lines` is the _old_ set of lines and `new_lines` is the new set. The iterator calls each of the lines from `new_lines` just `line` confusingly.
> 
> Basically I stand behind my block here.

I've renamed it for clarity.

Reply:
it will print `last_line` multiple times, but `last_line` is going to be different every time since we push to `lines` at the end of the closure. it has to be here, inside the closure to ensure that we override the spinner's message properly

Reply:
> If there are multiple new entries in `new_lines` this is gonna print `last_line` multiple times. I wouldn't mind if this fix as currently written was before the functional stuff instead of inside of it?

makes sense, I did not catch that. Just extracted it.